jetty-core/jetty-http2/jetty-http2-client-transport/src/main/java/org/eclipse/jetty/http2/client/transport/internal/HttpConnectionOverHTTP2.java

```diff
@@ -215,12 +219,40 @@ public void upgrade(Map<String, Object> context)
     protected void normalizeRequest(HttpRequest request)
     {
         super.normalizeRequest(request);
+        // Do not set the version explicitly, in case the
+        // request needs to be retried with a different version.
+        request.setVersion(HttpVersion.HTTP_2);
```

Comment:
The comment is unclear as it says _do not set the version explicitly_ but the next line does set the version explicitly.